internal/pkg/config/config.go

```diff
@@ -29,15 +29,47 @@ import (
 	"github.com/google/go-flow-levee/internal/pkg/config/regexp"
 )
 
-// FlagSet should be used by analyzers to reuse -config flag.
-var FlagSet flag.FlagSet
-var configFile string
+var (
+	// FlagSet should be used by analyzers to reuse -config flag.
+	FlagSet                      flag.FlagSet
+	configFile                   string
+	validFuncMatcherFields       []string
+	validSourceMatcherFields     []string
+	validFieldTagMatcherFields   []string
+	validFuncMatcherFieldSet     map[string]bool
+	validSourceMatcherFieldSet   map[string]bool
+	validFieldTagMatcherFieldSet map[string]bool
+)
 
 func init() {
 	FlagSet.StringVar(&configFile, "config", "config.yaml", "path to analysis configuration file")
+	validFuncMatcherFields = []string{
```

Comment:
What is the benefit of maintaining both a slice and a map? IMO we could just keep the slice. Also, what is the benefit of duplicating each field, as opposed to calling `strings.ToLower` before comparing, or comparing with `strings.EqualFold`? IMO doing that would be simpler and less error-prone.

Reply:
The slice is for error output and map for lookup. I didn't want to nest two for-loops when doing checking.

Reply:
Removed map and only use slice to compare.